### PR TITLE
DOC: add examples to np.char.multiply

### LINKED ISSUE #22267: DOC: `np.char` - list of functions that are missing examples – tracking issue
Find below the functions from `np.char` that are missing examples in the docstrings. If you have a suggestion for an example, please submit a pull request. 
To let everyone know that there is work in progress, link your PR to this issue. (If needed, refer to the [GitHub documentation](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue) on how to do it.)

- [x]  add
- [ ]  array
- [ ]     asarray
- [ ]     center
- [ ]     encode
- [ ]     equal
- [ ]     expandtabs
- [ ]     find
- [x]     greater
- [ ]     greater_equal
- [ ]     index
- [ ]     isalnum
- [ ]     isalpha
- [ ]     isdecimal
- [ ]     isdigit
- [ ]     islower
- [ ]     isnumeric
- [ ]     isspace
- [ ]     istitle
- [x]     isupper
- [ ]     join
- [ ]     less
- [ ]     less_equal
- [ ]     ljust
- [ ]     mod
- [ ]     multiply
- [ ]     not_equal
- [x]     partition
- [ ]     replace
- [ ]     rfind
- [ ]     rindex
- [ ]     rjust
- [x]     rpartition
- [ ]     rsplit
- [ ]     split
- [x]     splitlines
- [ ]     startswith
- [ ]     str_len
- [ ]     translate
- [ ]     zfill

### PR DESCRIPTION
**Pull request by Eric-Shang**
resolves #22267 partially by adding examples to np.char.multiply

### CODE REVIEW

File: numpy/core/defchararray.py [hunk 1]
```diff
@@ -333,7 +333,32 @@ def multiply(a, i):
...
+    For 2-D arrays:
+
+    >>> a = np.array(['a', 'b', 'c', 'd', 'e', 'f']).reshape((2,3))
+    >>> i = np.ones((2,3), np.int32) * 3
+    >>> i
+    array([[3., 3., 3.],
+           [3., 3., 3.]])
+    >>> np.char.multiply(a, i)
+    array([['aaa', 'bbb', 'ccc'],
+           ['ddd', 'eee', 'fff']], dtype='<U3')
```
Comment (rossbar) [inline numpy/core/defchararray.py:361]: Similar idea here, just a suggestion for simplifying - note that we can reuse the `i` defined above and simultaneously show broadcasting at work (without having to explicitly mention it):

```suggestion
    >>> a = np.array(['a', 'b', 'c', 'd', 'e', 'f']).reshape((2, 3))
    >>> np.char.multiply(a, 3)
    array([['aaa', 'bbb', 'ccc'],
           ['ddd', 'eee', 'fff']], dtype='<U3')
    >>> np.char.multiply(a, i)
    array([['a', 'bb', 'ccc'],
           ['d', 'ee', 'fff']], dtype='<U3')
```

**Review by rossbar — COMMENTED** [review]
Generally LGTM, thanks @Eric-Shang ! I've left a couple suggestions in an attempt to keep the docstring example as short as possible - I believe being concise has a lot of value.

Those are just suggestions though and not blockers!

**Comment by Eric-Shang** [comment]
> Generally LGTM, thanks @Eric-Shang ! I've left a couple suggestions in an attempt to keep the docstring example as short as possible - I believe being concise has a lot of value.
> 
> Those are just suggestions though and not blockers!

Great suggestions to learn!

**Comment by seberg** [comment]
Thanks for adding the example section @Eric-Shang, LGTM.

**Pull request merged**